src/cleanlab_codex/validation.py

```diff
@@ -0,0 +1,190 @@
+"""
+This module provides validation functions for checking if an LLM response is unhelpful.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable, Optional
+
+from cleanlab_codex.utils.prompt import default_format_prompt
+
+if TYPE_CHECKING:
+    from cleanlab_studio.studio.trustworthy_language_model import TLM  # type: ignore
+
+
+DEFAULT_FALLBACK_ANSWER = "Based on the available information, I cannot provide a complete answer to this question."
+DEFAULT_PARTIAL_RATIO_THRESHOLD = 70
+DEFAULT_TRUSTWORTHINESS_THRESHOLD = 0.5
+
+
+def is_bad_response(
+    response: str,
+    context: str,
+    tlm: TLM,  # TODO: Make this optional
+    query: Optional[str] = None,
+    # is_fallback_response args
+    fallback_answer: str = DEFAULT_FALLBACK_ANSWER,
+    partial_ratio_threshold: int = DEFAULT_PARTIAL_RATIO_THRESHOLD,
```

Comment:
missing optional `format_prompt` argument, which should be passed to `is_untrustworthy_response` if it was provided